src/evaluation/evaluator.h

```diff
@@ -1,312 +1,271 @@
-        m_startTime = std::chrono::system_clock::now();
+enum SearchType {
+    Default,
+    NullSearch,
+};
 
-        /*
-         * if a depth has been provided then make sure that we search to that depth
-         * timeout will be 10 minutes
-         */
-        if (depthInput.has_value()) {
-            m_endTime = m_startTime + std::chrono::minutes(10);
-        } else {
-            setupTimeControls(m_startTime, board);
-        }
+struct MoveResult {
+    uint64_t hash;
+    BitBoard board;
+};
 
-        m_hash = engine::generateHashKey(board);
+struct SearcherResult {
+    int32_t score;
+    uint8_t searcherId;
+    movegen::Move pvMove;
+    uint8_t searchedDepth;
+};
 
-        /* we're dependent on the hash table - ensure it's initialized! */
-        if (engine::TtHashTable::getSizeMb() == 0) {
-            engine::TtHashTable::setSizeMb(s_defaultTtHashTableSizeMb);
-        }
+template<uint8_t maxSize>
+class MoveVoteMap {
 
-        return scanForBestMove(depthInput.value_or(s_maxSearchDepth), board);
+    using MoveVotePair = std::pair<movegen::Move, uint8_t>;
+
+public:
+    MoveVoteMap() = default;
+
+    MoveVoteMap(std::initializer_list<MoveVotePair> init)
+    {
+        assert(init.size() < maxSize);
+
+        for (auto& entry : init) {
+            m_entries.at(m_size++) = std::move(entry);
+        }
     }
 
-    constexpr std::optional<movegen::Move> getPonderMove() const
+    constexpr uint8_t& at(const movegen::Move& ext_move)
     {
-        if (m_moveOrdering.pvTable().size() > 1) {
-            return m_moveOrdering.pvTable()[1];
+        for (auto& [move, vote] : m_entries) {
+            if (move == ext_move) {
+                return vote;
+            }
         }
+        assert(false);
 
-        return std::nullopt;
+        return m_entries.at(0).second;
     }
 
-    /* TODO: ensure time handler is stopped as well - not just assumed stopped. This is racy */
-    constexpr void stop()
+    constexpr bool count(const movegen::Move& ext_move)
     {
-        m_isStopped = true;
+        for (auto& [move, vote] : m_entries) {
+            if (move == ext_move) {
+                return true;
+            }
+        }
+        return false;
     }
 
-    constexpr uint64_t getNodes() const
+    constexpr void insert(movegen::Move move, uint8_t vote)
     {
-        return m_nodes;
+        assert(m_size < maxSize);
+        m_entries.at(m_size++) = MoveVotePair(move, vote);
     }
 
-    constexpr void printEvaluation(const BitBoard& board, std::optional<uint8_t> depthInput = std::nullopt)
+    constexpr void clear()
     {
-        resetTiming(); /* to reset nodes etc but not tables */
+        for (uint8_t i = 0; i < m_size; i++) {
+            m_entries.at(i) = MoveVotePair {};
+        }
+        m_size = 0;
+    }
 
-        uint8_t depth = depthInput.value_or(5);
+    auto* begin() { return m_entries.begin(); }
 
-        fmt::println("");
+    auto* end() { return m_entries.begin() + m_size; }
```

Comment:
For `*this` to work we need to add const iterators as well:

```suggestion
    auto* end() { return m_entries.begin() + m_size; }
    
    auto* begin() const { return m_entries.begin(); }
    auto* end() const { return m_entries.begin() + m_size; }
```

Reply:
Done.